### DISCUSSION TITLE
How to I wire up my own test framework to the in-browser test support?

### DISCUSSION THREAD

**Discussion by NullVoxPopuli** [vitest-dev/vitest#3428] [Q&A]
If I have my own custom test framework, what do I need to do in order to communicate with the vitest CLI? (because a test framework needs to report progress to the CLI for CI usage).

thanks!

**Comment by sheremet-va** [comment]
I don't think there is a good way to do this. If you don't use `@vitest/runner`, I don't really know why you would choose Vitest for this.

Here is the file that does browser-node communication: 

https://github.com/vitest-dev/vitest/blob/41e11dad39d523af0aa7b50d22b2b9c6dd980a7b/packages/browser/src/client/main.ts

**Reply by NullVoxPopuli** [reply]
my interest is mostly in the fact that there is a gap in available tooling htat allows for:
- vite does everything it does (compile, serve, etc)
- bring your own test framework
- ability to do your own test reporting from the visited page hosted by vite
